Source/UI/NewUI/PaymentForm/Model/PaymentFormConfiguration.swift

```diff
@@ -24,11 +25,11 @@ public struct PaymentFormConfiguration {
      */
     public init(apiKey: String,
                 environment: Environment,
-                supportedSchemes: [Card.Scheme],
+                supportedSchemes: [CardScheme],
                 billingFormData: BillingForm?) {
         self.serviceAPIKey = apiKey
         self.environment = environment
-        self.supportedSchemes = supportedSchemes
+        self.supportedSchemes = supportedSchemes.compactMap { $0.mapToCheckoutCardScheme() }
```

Comment:
May have been left in by mistake. No need to remove optionals here
```suggestion
        self.supportedSchemes = supportedSchemes.map { $0.mapToCheckoutCardScheme() }
```

Reply:
The first one is a habit, I think wherever you do a map, making it compactMap is good practice as it future proofs your expectation of not having nils. Unless if you have a counter argument, I'm still preferring it as good practice (don't go down the optimisation argument, I'd just say the difference makes it an over-optimisation - nice to have if needed; large sets of expensive operation at some key point, not UI triggered for less then 100 possible elements) :).

Second one, I'm not a huge fan of computed properties for enabling this, as I find them to uglify (if that's a word) the syntax inside the object. They're useful sometimes and absolutely should be used, but wouldn't do just for something like this.

Reply:
Disagree with the first point. We should be minimising the optionals in our codebase for the sake of simplicity. As a reader, compactMap infers that supportedSchemes is type [CardScheme?], which is quite confusing. Same way you wouldn't have optionals all over your models if the values were always present.

Reply:
> We should be minimising the optionals in our codebase for the sake of simplicity.

💯 Agree, but would point out its not always possible and there are genuine times you just need to handle optionals in different shapes and forms.

***

> As a reader, compactMap infers that supportedSchemes is type [CardScheme?], which is quite confusing

Not necessarily, as you can always check the type and see that is not true. It simply keeps it working if for some reason in the future you do need to handle optionals in that source array.

*** 

Found this nice comparison of [map vs flatMap vs compactMap](https://www.hackingwithswift.com/articles/205/whats-the-difference-between-map-flatmap-and-compactmap)

> map() will take a value out of its container, transform it using the code you specify, then put it back in its container. compactMap() does the same thing, but if your transformation returns an optional it will be unwrapped and have any nil values discarded.

To me this syntax just better protects the code as we know we absolutely want to have a result without optionals. If you end up changing the data source, that's a separate problem which you'll need looking at in isolation, but it will keep this assignment guaranteed to not handle optionals.